From 82d770339af9daed2a641d44db34cd83d5d8d470 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:19 -0400
Subject: [tools] perf/trace: Beautify rt_sigprocmask 'how' arg

Message-id: <1408462094-14194-91-git-send-email-jolsa@redhat.com>
Patchwork-id: 88074
O-Subject: [PATCH RHEL7.1 BZ1131394 090/325] perf trace: Beautify rt_sigprocmask 'how' arg
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit eb5b1b1475d5ce6ebe84fd4c641545a41839d5f9
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Sep 3 16:37:46 2013 -0300
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index cc57f41..df9a66c 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -252,6 +252,9 @@ static const char *fcntl_cmds[] = {
 };
 static DEFINE_STRARRAY(fcntl_cmds);
 
+static const char *sighow[] = { "BLOCK", "UNBLOCK", "SETMASK", };
+static DEFINE_STRARRAY(sighow);
+
 static size_t syscall_arg__scnprintf_open_flags(char *bf, size_t size,
             struct syscall_arg *arg)
 {
@@ -366,6 +369,9 @@ static struct syscall_fmt {
  { .name     = "pwrite",     .errmsg = true, .alias = "pwrite64", },
  { .name     = "read",     .errmsg = true, },
  { .name     = "recvfrom",   .errmsg = true, },
+ { .name     = "rt_sigprocmask", .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_STRARRAY, /* how */ },
+   .arg_parm  = { [0] = &strarray__sighow, /* how */ }, },
  { .name     = "select",     .errmsg = true, .timeout = true, },
  { .name     = "setitimer",  .errmsg = true,
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* which */ },
-- 
1.7.1